src/python/foglamp/data_purge/sqlalchemy_purge.py

```diff
@@ -0,0 +1,324 @@
+"""
+Description: Based on FOGL-200 (https://docs.google.com/document/d/1GdMTerNq_-XQuAY0FJNQm09nbYQSjZudKJhY8x5Dq8c/edit) 
+    the purge process is suppose to remove data based on either a user_id, or an X amount of time back depending on
+    whether or not the configuration (config.json) requires to retain data that has not been sent to the Pi System.
+    
+    Given that the code is dependent on configuration files, sending data to Pi, and connecting to the database, 
+    I have "hard-coded" those dependencies with the use of extra methods, and files. This includes things like: 
+    get_nth_id which updates the config file with a 'random' last ID that was sent to Pi, and the main which also 
+    as a scheduler process. 
+    
+     Specifically the purge process (purge_process_function) does the following: 
+     - Based on the configuration file (retainUnsent) it either removes by the lastID 
```

Comment:
No, if retainUnsent is set to true it should still respect age and not remove readings newer than the defined age even f they have been sent too PI